pkg/ds/fields/fields.go

```diff
@@ -0,0 +1,117 @@
+package fields
+
+import (
+	"encoding/json"
+
+	"k8s.io/kubernetes/pkg/labels"
+
+	"github.com/square/p2/pkg/pods"
+	"github.com/square/p2/pkg/types"
+)
+
+// ID is a named type alias for Daemon Set IDs
+type ID string
+
+// Cluster name is where the daemon set lives on
+type ClusterName string
+
+// DaemonSet holds the runtime state of a Daemon Set as saved in Consul.
+type DaemonSet struct {
+	// UUID for this controller
+	ID ID
+
+	// When disabled, this controller will not make any scheduling changes
+	Disabled bool
+
+	// The pod manifest that should be scheduled on nodes
+	Manifest pods.Manifest
+
+	// Minimum health for nodes when scheduling
+	MinHealth int
+
+	// Daemon Set's environment name
+	Name ClusterName
+
+	// Defines the set of nodes on which the manifest can be scheduled
+	NodeSelector labels.Selector
+
+	// PodID to deploy
+	PodID types.PodID
+}
+
+// RawDaemonSet defines the JSON format used to store data into Consul
+type RawDaemonSet struct {
+	ID           ID          `json:"id"`
+	Disabled     bool        `json:"disabled"`
+	Manifest     string      `json:"manifest"`
+	MinHealth    int         `json:"MinHealth"`
```

Comment:
`min_health` would match the other fields. Is there a good reason for this to be camelcased?